src/fios_gui.cpp

```diff
@@ -864,6 +883,19 @@ static WindowDesc _save_dialog_desc(
 	_nested_save_dialog_widgets, lengthof(_nested_save_dialog_widgets)
 );
 
+/**
+ * Callback function for the savegame 'are you sure you want to overwrite save' window
+ * @param w Window which is calling this callback
+ * @param confirmed boolean value, true when yes was clicked, false otherwise
+ */
+void SaveGameConfirmationCallback(Window *w, bool confirmed)
+{
+	if (confirmed) {
+		_switch_mode = SM_SAVE_GAME;
+		FiosMakeSavegameName(_file_to_saveload.name, dynamic_cast<SaveLoadWindow*>(w)->filename_editbox.text.buf, lastof(_file_to_saveload.name));
```

Comment:
maybe an assert to check that the dynamic_cast succeeded? (with use of a temporary variable)